cmd/intoto-custom-predicate/attest.go

```diff
@@ -158,50 +164,89 @@ run in the context of a Github Actions workflow.`,
 			check(err)
 
 			if attPath != "" {
-				s := sigstore.NewDefaultSigner()
-				att, err := s.Sign(ctx, p)
-				check(err)
+				if local {
+					// use local version
+					s := signing.NewDefaultFulcio()
+					att, err := s.Sign(ctx, p)
+					check(err)
 
-				_, err = s.Upload(ctx, att)
-				check(err)
+					r := sigstore.NewDefaultRekor()
 
-				f, err := getFile(attPath)
-				check(err)
+					_, err = r.Upload(ctx, att)
+					check(err)
 
-				_, err = f.Write(att.Bytes())
-				check(err)
+					f, err := getFile(attPath)
+					check(err)
+
+					_, err = f.Write(att.Bytes())
+					check(err)
+				} else {
+
+					_, err = github.NewOIDCClient()
+					check(err)
+
+					s := sigstore.NewDefaultFulcio()
+					att, err := s.Sign(ctx, p)
+					check(err)
+
+					r := sigstore.NewDefaultRekor()
+
+					_, err = r.Upload(ctx, att)
+					check(err)
+
+					f, err := getFile(attPath)
+					check(err)
+
+					_, err = f.Write(att.Bytes())
+					check(err)
```

Comment:
can the upload, getfile and write be done outside the if block? looks like it can be de-duplicated